pkg/cluster/spec/instance.go

```diff
@@ -272,11 +281,19 @@ func (i *instance) LogDir() string {
 }
 
 func (i *instance) OS() string {
-	return reflect.ValueOf(i.InstanceSpec).FieldByName("OS").Interface().(string)
+	v := reflect.ValueOf(i.InstanceSpec).FieldByName("OS")
```

Comment:
I don't think there should be a component that hasn't `OS/Arch` (`any` is an acceptable value).

Reply:
If it has these fields, the global one will be used, and then it requires user to specify `any` in topology file

Reply:
You can use `OS string default:any` to avoid inherit global value.